### ISSUE TITLE
jsdoc/valid-types Rule Errors when namepath contains `readonly` (all lowercase)

### ISSUE THREAD

**Issue by dhurlburtusa** [gajus/eslint-plugin-jsdoc#813]
## Expected behavior

The following should be valid and should not generate an error.

```js
/**
 * @param {object} params
 * @param {boolean} params.readonly
 */
```

## Actual behavior

```js
/**
 * @param {object} params
 * @param {boolean} params.readonly
 */
```

Errors with: "Syntax error in namepath: params.readonly  jsdoc/valid-types".

## ESLint Config

```js
module.exports = {
  root: true,
  extends: ["plugin:jsdoc/recommended"],
  rules: {
    "jsdoc/valid-types": "error",
  },
};
```

## ESLint sample

```js
/**
 * @param readonly
 */
```

```js
/**
 * @param {boolean} readonly
 */
```

```js
/**
 * @param {object} params
 * @param {boolean} params.readonly
 */
```

## Environment

- Node version: 16.6.1
- ESLint version: 8.4.1
- `eslint-plugin-jsdoc` version: 37.2.0

## Possible Cause

I believe the cause of this is due to the following PR: https://github.com/jsdoc-type-pratt-parser/jsdoc-type-pratt-parser/pull/98

Although the cause may be due to the parser (I filed an issue there too (https://github.com/jsdoc-type-pratt-parser/jsdoc-type-pratt-parser/issues/104)), I'm thinking it would be good to improve the tests here to help catch some of these errors.

**Comment by brettz9** [comment]
I'd prefer we get any tests added to `jsdoc-type-pratt-parser`. Don't need to cover every edge case in this project, I think. Closing but feel free to comment further as needed.

**Comment by jdatskuid** [comment]
I ran into this error today in a typedef. We have an object with a property named "readonly" and I get this error.

Example:

```javascript
/**
 * An object interface
 * @typedef  {Object} FooBar
 * @property {boolean} readonly
 * @property {boolean} private
 * @property {boolean} public
 * @property {boolean} constant
 */
```

Error:

```
32:0  error  Syntax error in namepath: readonly          jsdoc/valid-types
```

**Comment by brettz9** [comment]
The issue is being tracked at [`jsdoc-type-pratt-parser`.](https://github.com/jsdoc-type-pratt-parser/jsdoc-type-pratt-parser/issues/104).

You might try surrounding the item in quotes and see if that provides a temporary resolution.

**Comment by jdatskuid** [comment]
Thanks, I noticed that after posting. Quotes do not resolve the issue.
